### ISSUE TITLE
Loading Spinner without WCAG accessibility certification

### ISSUE THREAD

**Issue by UBOrange** [telekom/scale#600]
According to Scale (https://www.brand-design.telekom.com/scale/?path=/usage/beta-components-loading-spinner--standard) the "Loading Spinner" is a "Beta Component" without WCAG accessibility certification.
We actually use the "Loading Spinner" component.
The recent official accessibility report created for our application by an external appraiser stated this as an issue "to be solved" (user who cannot see the spinner do not get any information about the fact that the spinner is running).

Question to the Scale team: By when could we expect that the "Loading Spinner" component will be adapted and will get its WCAG accessibility certification?

**Comment by acstll** [comment]
Thanks for asking, @UBOrange

The Loading Spinner component got the AA certification in the past few days (in a feature branch), and a new release to npm should be imminent. I will let you know here when that's available.

**Comment by nowseemee** [comment]
Hi @UBOrange - we just released the new version of the Loading Spinner which is accessible - `3.0.0-beta.21`. Storybook deployment will follow later.

**Comment by UBOrange** [comment]
**@acstll** - We just checked whether the Loading Spinner in Version 3.0.0-beta.21 would solve our problem. Unfortunately, it does not solve the problem. We checked this with the screen reader NVDA. It does not tell the (blind) user that the spinner is running and why it is running.
Could you please re-open the ticket and find a solution?
Thanks a lot! :-)

**Comment by acstll** [comment]
Thank you @UBOrange for the feedback.

It is possible that you need to change your markup a bit to solve the issue. Would you be so kind and share a snippet of your code with some context? This way we can compare with what we know.

We use JAWS2019 on Windows 10 for screen reader testing. I'm thinking there's a possibility that NVDA yields different results too. But first let's not get ahead and first take a look at the code.

Thanks again.

**Comment by ChristianAtWork** [comment]
Some Code:
![image](https://user-images.githubusercontent.com/43496354/135809109-fe5ca8e2-439f-422f-b8eb-8700feeafd14.png)

![image](https://user-images.githubusercontent.com/43496354/135809362-85db81bf-5668-4f27-97be-f94290bbc804.png)

**Comment by acstll** [comment]
Thank you @ChristianAtWork, we'll get back to you.

**Comment by acstll** [comment]
Hey @ChristianAtWork, thanks for waiting!

We believe there's a delta in the way NVDA works versus other screen readers.

The current implementation of the Loading Spinner reads the text in the `text` attribute both in JAWS 2019 (the screen reader we use for testing officially), and also macOS VoiceOver. This is done with an element with the `aria-live="polite"` attribute.

According to this https://github.com/nvaccess/nvda/issues/7906#issuecomment-807957477 NVDA won't read that on load, but only if the `textContent` of the element with the `aria-live` attribute changes.

So I would suggest two solutions for you to try.

## Adding an extra element to your `Spinner` component

The extra element would feature the text, an [sr-only-type of class](https://kittygiraudel.com/snippets/sr-only-class/) to hide it visually, and `role="alert"` so NDVA would read it. (An `SrOnly` component would make sense here maybe, if you don't have global CSS classes.)

(I'm omitting most props for brevity…)

```jsx
<div className={classNames}>
  <ScaleLoadingSpinner text={text} />
  <div className="sr-only" role="alert">{text}</div>
</div>
```

## Setting the `text` prop to `''` first

…and then changing it to the actual text after a few milliseconds using some React hooks. This I'm not sure it would work and it's definitely more complex than the first option, but I think it's worth mentioning.

---

Is it possible for you to implement any of this and have the accessibility team test it?

Hope this helps, looking forward to any more feedback.

**Comment by UBOrange** [comment]
Thank you very much for your help :-)
 
<div className="sr-only" role="alert">{text}</div>

solved the problem.
